pkg/scheduler/objects/application.go

```diff
@@ -827,13 +827,41 @@ func (sa *Application) tryPlaceholderAllocate(nodeIterator func() NodeIterator,
 			if ph.released || request.taskGroupName != ph.taskGroupName {
 				continue
 			}
+			// before we check anything we need to check the resources equality
+			delta := resources.Sub(ph.AllocatedResource, request.AllocatedResource)
```

Comment:
Maybe we could replace "ph" (placeholder) with "hold" (v.s. "request") to make our codebase easier to understand. :) But if it appears in too many places, it's okay.

Reply:
too big a change